### DISCUSSION TITLE
Aims for next releases?

### DISCUSSION THREAD

**Discussion by mwtoews** [Toblerity/Fiona#1074] [General]
There are currently three active branches of development: [`maint-1.8`](https://github.com/Toblerity/Fiona/tree/maint-1.8), [`maint-1.9`](https://github.com/Toblerity/Fiona/tree/maint-1.9), and [`master`](https://github.com/Toblerity/Fiona/tree/master) (currently 2.0dev). I have a few questions for these.

maint-1.8 is a maintenance branch for the 1.8.x series, with [1.8.21](https://github.com/Toblerity/Fiona/releases/tag/1.8.21) released in Feb 2022. It supports Python 2.6+ thru to Python 3.10. Has maintenance for this branch come to an end?

maint-1.9 is a branch for a 1.9.x series, which has no releases or release candidates. It currently supports Python 2.6+ thru to Python 3.10. What are the aims for this release?

master is the development branch for 2.0.x with new features not found in other branches. It finally drops Python 2 support, requiring Python 3.6+. Why focus on the unreleased 1.9.x series, and not efforts for 2.0.x based on master?

**Comment by sgillies** [comment]
@mwtoews thank you for asking about this!

The main development branch (master, aka Fiona 2.0) will have breaking changes. Those changes are elaborated in https://github.com/Toblerity/fiona-rfc/blob/master/rfc/0001-fiona-2-0-changes.md. The RFC is a little dated and needs some updates. It's from a time before GitHub discussions. We could probably do that kind of RFC thing here now, or at least do the discussion of the RFCs here.

When that was written, the current Fiona release was 1.8.something. We thought it best to have a 1.9.0 release that would warn about breaking changes and pave the way for 2.0. The changes in 2.0 won't just involve Python versions. I'm proposing to replace fiona's GeoJSON feature-like dicts with extension classes. It'll be a big deal.

Frankly, I bungled the branching. The branch we call "maint-1.9" should be the main branch (the next release) and the work on 2.0 should be in a different feature branch. This has caused me a lot of pain over the past ~2 years. Currently, however, the maint-1.8, maint-1.9, and master branches are pretty well sorted out. 1.8.21 was merged to maint-1.9 and then maint-1.9 was merged to master. The master branch has all the bug fixes of the other two.

I still believe that a 1.9.0 release benefits users of the package. We're going to do it, but then I will very quickly switch over to work on 2.0.

**Comment by mwtoews** [comment]
Thanks @sgillies ! I missed that RFC, so it explains lots, including why there is a 1.9 before 2.0. If that RFC is still being drafted/updated, I'd also like to see dropping OrderedDict, as CPython 3.6+ dict objects [preserve their insertion order](https://stackoverflow.com/q/39980323/).

A few other questions:

- Will the minimum Python version for 1.9 change? (e.g. 2.6->2.7). I don't have any big issue with the Python 2 stuff, but it helps shape PRs that assume a minimum Python version, e.g. #985
- Should changes/PRs that don't break changes be backported from `master` to `maint-1.9`? I'm guessing yes.

**Reply by sgillies** [reply]
The minimum Python version for 1.9.0 should adapt to reality. It's difficult to use 2.7 now. We might even require Python >= 3.6.

I'm of the mind that we shouldn't backport anything but fixes for show-stopping bugs to maint-1.9 so that we don't get bogged down in merge conflicts. I'd like to get 1.9.0 done quickly.

**Reply by mwtoews** [reply]
Thanks again. I can look at implementing a minimum Python version change for `maint-1.9` in the coming weeks, bearing in mind about avoiding merge conflicts with `master`. But yes, I can chip in some time to get a 1.9.0 out quickly!

**Reply by mwtoews** [reply]
xref #1079
